libs/assets/src/lib/asset-details-dialog.tsx

```diff
@@ -19,20 +19,39 @@ import create from 'zustand';
 export type AssetDetailsDialogStore = {
   isAssetDetailsDialogOpen: boolean;
   assetDetailsDialogSymbol: string | Asset;
+  assetDetailsDialogTrigger: HTMLElement | null | undefined;
   setAssetDetailsDialogOpen: (isOpen: boolean) => void;
   setAssetDetailsDialogSymbol: (symbol: string | Asset) => void;
+  setAssetDetailsDialogTrigger: (
+    trigger: HTMLElement | null | undefined
+  ) => void;
+  openAssetDetailsDialog: (
+    symbol: string | Asset,
+    trigger?: HTMLElement | null
+  ) => void;
 };
 
 export const useAssetDetailsDialogStore = create<AssetDetailsDialogStore>(
   (set) => ({
     isAssetDetailsDialogOpen: false,
     assetDetailsDialogSymbol: '',
-    setAssetDetailsDialogOpen: (isOpen: boolean) => {
+    assetDetailsDialogTrigger: null,
+    setAssetDetailsDialogOpen: (isOpen) => {
```

Comment:
chore (non blocking): I refactored the global store to just use a single update function which should be sufficient here too.